### PR TITLE
Docs: Add backup and restore details

### PR DESCRIPTION
**Pull request by oraNod**
issue #1212 

Needs backport to 2.1.x.

### CODE REVIEW

File: documentation/asciidoc/topics/proc_restoring_cluster.adoc [hunk 1]
```diff
@@ -42,13 +41,14 @@ include::yaml/backup_resources.yaml[]
 ----
 $ {oc_apply_cr} my-restore.yaml
 ----
-+
-A new pod joins the {brandname} cluster and restores state from the backup file.
-When the operation is complete, the pod leaves the cluster and logs the following message:
+
+.Verification
+
+. Check {brandname} logs for the following message:
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_restoring_cluster.adoc:47]: We should probably tell the user to check the `Restore` CR status as well. In the event of a successful restore, the `.status.phase` will be set to `Succeeded`.

File: documentation/asciidoc/topics/yaml/backup.yaml [hunk 2]
```diff
@@ -4,3 +4,6 @@ metadata:
   name: my-backup
 spec:
   cluster: source-cluster
+  volume:
+    storage: 1Gi
+    storageClassName: "blah"
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/yaml/backup.yaml:9]: You forgot :stuck_out_tongue_winking_eye:

File: documentation/asciidoc/topics/proc_restoring_cluster.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/asciidoc/topics/proc_restoring_cluster.adoc:47]: Is this true for `Backup` CR as well?

File: documentation/asciidoc/topics/yaml/backup.yaml (hunk 2 above)
Comment (oraNod) [reply documentation/asciidoc/topics/yaml/backup.yaml:9]: nuts. the one thing...

File: documentation/asciidoc/topics/proc_restoring_cluster.adoc (hunk 1 above)
Comment (ryanemerson) [reply documentation/asciidoc/topics/proc_restoring_cluster.adoc:47]: Yep

Comment (oraNod) [reply documentation/asciidoc/topics/proc_restoring_cluster.adoc]: maybe we should create a new procedure for handling failed backup/restore operations?

Comment (ryanemerson) [reply documentation/asciidoc/topics/proc_restoring_cluster.adoc]: Do you mean document how to handle failures? Or implement something in the operator? I'm definitely in favour of the former.

Comment (oraNod) [reply documentation/asciidoc/topics/proc_restoring_cluster.adoc]: Exactly, we should document how to handle failures. I'll update shortly with a topic we can work through.

**Comment by oraNod** [comment]
@ryanemerson Seems like there's some weird stuff going on with GitHub as this PR doesn't have the same commits as my branch: https://github.com/oraNod/infinispan-operator/tree/issue-1212 

Think it's just a service disruption but FYI not to merge until this gets sorted.

**Pull request closed without merging**